API to Retrieve Experts by Keywords

noweber opened the pull request:
This PR adds an API with basic functionality to retrieve a set of experts by a keywords string.

noweber committed:
Created REST API Shell with Experts Endpoint for Search and Collaborators

This commit adds a temp

noweber committed:
Updated Project for Swagger UI in All Environments and Hosting

noweber committed:
Added API Pagination Interfaces for Getting Experts and Collaborators

noweber committed:
Lucene Filter Query Parameter for Experts

noweber committed:
Added Covid Experts for M2 Presentation

noweber committed:
Basic Elasticsearch Querying to Localhost

noweber committed:
Elasticsearch Environment Settings

This commit refactors the code to get the Elasticsearch URL, username, and password from configuration files so that this can be retrieved in production as an environment variable.

noweber committed:
Removing Unnecessary Files

This commit removes unnecessary bin and obj files from the repository and the gitignore will not add them later.

noweber committed:
Basic Publication and Relevancy Sum Metrics

This commit adds basic relevancy and publication count sums.

noweber committed:
Configurable Elasticsearch Query Limit

This commit adds a configurable value for the number of records to return from Elasticsearch queries.

noweber committed:
Added Configurable Application Insights Logging

This commit adds Application Insights logging via configurations in the app settings.

noweber committed:
Added Pagination to Experts Querying

This commit adds pagination to the expert search using query parameters.

noweber committed:
Added CI/CD via GitHub Actions

noweber committed:
Corrected Page Sorting Orders

This commit corrects an issue where the page sorting was in ascending order by default instead of descending.

noweber committed:
Fixed Null Name Bug

This commit fixes a bug where null names would end up being pooled as the same aggregation metric.

noweber committed:
Added API Setup Instructions to the README File

noweber committed:
Added API URL to the README

noweber committed:
Added Ascending/Descending Sorting for Getting Experts by Keywords

This commit adds ascending and descending sorts to the GET experts by keywords endpoint.

pagreene reviewed (approved):
Looks like you have some collisions in the readme and gitignore that you will have to resolve, but aside from those relative trivialities, I'm afraid I'm not well qualified to review C# or .NET. For the sake of moving things along though, I will trust that you are able to self-monitor well enough for our purposes at this juncture. So you have my github-satisfying blessing to merge when you see fit. :)

pagreene commented:
Oh, and remember to select "rebase and merge", especially if you want your contributions to be accurately represented, and to preserve documentation of the work done.

noweber committed:
Added Collaborators and Time Range Querying

This commit adds functionality to get a user's collaborator set and allows the request to be filtered within a time range.

noweber committed:
Added Citations Count in Expert Metrics

This commit adds citations count to the expert metrics. This looks at each expert's publications and then determines how many times all of their publications are cited by other experts within the context of a keywords search. It also provide the ability to sort the results by citations count (and changes citations count to be the default sorting metric).

noweber committed:
Merge branch 'main' into main
